dear_petition/petition/helpers.py

```diff
@@ -24,3 +29,36 @@ def get_285_form_agency_address(agency):
         body += f"{field}\n"
 
     return body
+
+
+def get_text_pixel_length(text):
+    font = ImageFont.truetype(str(settings.APPS_DIR.path("static/times.ttf")), 12)
+    size = font.getsize(text)
+    return size[0]
+
+
+def get_truncation_point_of_text_by_pixel_size(text, desired_length):
+    """
+    Given a string and a desired pixel length, will return the index the string needs to be truncated to obtain the the maximal string that can fit within the desired length
+    """
+    letter_lengths = {}
+
+    def calculate_letter_length(letter):
+        if letter in letter_lengths:
+            return letter_lengths[letter]
+        letter_length = get_text_pixel_length(letter)
+        letter_lengths[letter] = letter_length
+        return letter_length
+
+    truncated_string_size = 0
+    idx = 0
+    while True:
```

Comment:
I think we need to add a check for the length of `text`, otherwise we might hit an infinite loop when the string fits within the `desired_length`.

```suggestion
    while idx < len(text):
```